py/visdom/server.py

```diff
@@ -887,15 +894,43 @@ def initialize(self, app):
         self.state = app.state
         self.port = app.port
         self.env_path = app.env_path
+        self.with_login = app.with_login
+        self.user_login = app.user_login
 
     def get(self, args, **kwargs):
         items = gather_envs(self.state, env_path=self.env_path)
-        self.render(
-            'index.html',
-            user=getpass.getuser(),
-            items=items,
-            active_item=''
-        )
+
+        if (not self.with_login) or self.with_login and self.current_user:
```

Comment:
Where is `self.current_user` set? Is this handled by tornado? As this is somewhat unclear (it isn't set in this file anywhere), please add a comment about where it's coming from for future contributors 👍.